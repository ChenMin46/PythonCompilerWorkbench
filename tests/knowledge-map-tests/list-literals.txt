[]
[1]
[1, 2, 3]
[1, 2, 3, "hello"]
[x, y + z]
[1, [2, 3]]
[1, 2] + [3, 4]
[1, 2] * 3

# multiline
[1,
      2,
  3       ,   
  
        "hello"]
