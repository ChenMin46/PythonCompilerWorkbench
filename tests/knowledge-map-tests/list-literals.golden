[]
[1]
[1, 2, 3]
[1, 2, 3, "hello"]
[x, y + z]
[1, [2, 3]]
[1, 2] + [3, 4]
[1, 2] * 3

# multiline
[1,
      2,
  3       ,   
  
        "hello"]
===
Module
  body: [
    Expr
      value: List >> '[]'
        elts: [
        ]
        ctx: Load
    Expr
      value: List >> '[1]'
        elts: [
          Num >> '1'
            n: 1
        ]
        ctx: Load
    Expr
      value: List >> '[1, 2, 3]'
        elts: [
          Num >> '1'
            n: 1
          Num >> '2'
            n: 2
          Num >> '3'
            n: 3
        ]
        ctx: Load
    Expr
      value: List >> '[1, 2, 3, "hello"]'
        elts: [
          Num >> '1'
            n: 1
          Num >> '2'
            n: 2
          Num >> '3'
            n: 3
          Str >> '"hello"'
            s: 'hello'
        ]
        ctx: Load
    Expr
      value: List >> '[x, y + z]'
        elts: [
          Name >> 'x'
            id: 'x'
            ctx: Load
          BinOp >> 'y + z'
            left: Name >> 'y'
              id: 'y'
              ctx: Load
            op: Add
            right: Name >> 'z'
              id: 'z'
              ctx: Load
        ]
        ctx: Load
    Expr
      value: List >> '[1, [2, 3]]'
        elts: [
          Num >> '1'
            n: 1
          List >> '[2, 3]'
            elts: [
              Num >> '2'
                n: 2
              Num >> '3'
                n: 3
            ]
            ctx: Load
        ]
        ctx: Load
    Expr
      value: BinOp >> '[1, 2] + [3, 4]'
        left: List >> '[1, 2]'
          elts: [
            Num >> '1'
              n: 1
            Num >> '2'
              n: 2
          ]
          ctx: Load
        op: Add
        right: List >> '[3, 4]'
          elts: [
            Num >> '3'
              n: 3
            Num >> '4'
              n: 4
          ]
          ctx: Load
    Expr
      value: BinOp >> '[1, 2] * 3'
        left: List >> '[1, 2]'
          elts: [
            Num >> '1'
              n: 1
            Num >> '2'
              n: 2
          ]
          ctx: Load
        op: Mult
        right: Num >> '3'
          n: 3
    Expr
      value: List ML>> '[1,\n      2,\n  3       ,   \n  \n        "hello"]'
        elts: [
          Num >> '1'
            n: 1
          Num >> '2'
            n: 2
          Num >> '3'
            n: 3
          Str >> '"hello"'
            s: 'hello'
        ]
        ctx: Load
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                "[]"
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_1", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_2", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_3", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_4", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "2"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_5", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_6", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_7", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_8", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "2"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_9", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_10", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    "\"hello\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }, 
                {
                  "id": "tid_11", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_12", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "x"
                  ], 
                  "id": "id_0", 
                  "name": "Name"
                }, 
                {
                  "id": "tid_13", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "contents": [
                        "y"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }, 
                    " ", 
                    {
                      "id": "tid_14", 
                      "name": "token", 
                      "value": "+"
                    }, 
                    " ", 
                    {
                      "contents": [
                        "z"
                      ], 
                      "id": "id_0", 
                      "name": "Name"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp"
                }, 
                {
                  "id": "tid_15", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_16", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_17", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "id": "tid_18", 
                      "name": "token", 
                      "value": "["
                    }, 
                    {
                      "contents": [
                        "2"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_19", 
                      "name": "token", 
                      "value": ","
                    }, 
                    " ", 
                    {
                      "contents": [
                        "3"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_20", 
                      "name": "token", 
                      "value": "]"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "List"
                }, 
                {
                  "id": "tid_21", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    {
                      "id": "tid_22", 
                      "name": "token", 
                      "value": "["
                    }, 
                    {
                      "contents": [
                        "1"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_23", 
                      "name": "token", 
                      "value": ","
                    }, 
                    " ", 
                    {
                      "contents": [
                        "2"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_24", 
                      "name": "token", 
                      "value": "]"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "List"
                }, 
                " ", 
                {
                  "id": "tid_25", 
                  "name": "token", 
                  "value": "+"
                }, 
                " ", 
                {
                  "contents": [
                    {
                      "id": "tid_26", 
                      "name": "token", 
                      "value": "["
                    }, 
                    {
                      "contents": [
                        "3"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_27", 
                      "name": "token", 
                      "value": ","
                    }, 
                    " ", 
                    {
                      "contents": [
                        "4"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_28", 
                      "name": "token", 
                      "value": "]"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "List"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    {
                      "id": "tid_29", 
                      "name": "token", 
                      "value": "["
                    }, 
                    {
                      "contents": [
                        "1"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_30", 
                      "name": "token", 
                      "value": ","
                    }, 
                    " ", 
                    {
                      "contents": [
                        "2"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_31", 
                      "name": "token", 
                      "value": "]"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "List"
                }, 
                " ", 
                {
                  "id": "tid_32", 
                  "name": "token", 
                  "value": "*"
                }, 
                " ", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n\n", 
        {
          "id": "tid_33", 
          "name": "comment", 
          "value": "# multiline"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "tid_34", 
                  "name": "token", 
                  "value": "["
                }, 
                {
                  "contents": [
                    "1"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_35", 
                  "name": "token", 
                  "value": ","
                }, 
                "\n      ", 
                {
                  "contents": [
                    "2"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                {
                  "id": "tid_36", 
                  "name": "token", 
                  "value": ","
                }, 
                "\n  ", 
                {
                  "contents": [
                    "3"
                  ], 
                  "id": "id_0", 
                  "name": "Num"
                }, 
                "       ", 
                {
                  "id": "tid_37", 
                  "name": "token", 
                  "value": ","
                }, 
                "   \n  \n        ", 
                {
                  "contents": [
                    "\"hello\""
                  ], 
                  "id": "id_0", 
                  "name": "Str"
                }, 
                {
                  "id": "tid_38", 
                  "name": "token", 
                  "value": "]"
                }
              ], 
              "id": "id_0", 
              "name": "List"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
